toolchain/sem_ir/typed_insts_test.cpp

```diff
@@ -31,14 +31,17 @@ namespace {
   static_assert(Name::Kind == InstKind::Name);
 #include "toolchain/sem_ir/inst_kind.def"
 
-template <typename TypedInst>
-auto CommonFieldOrder() -> void {
-  Inst inst = InstTestHelper::MakeInst(TypedInst::Kind, Parse::Node(1),
-                                       TypeId(2), 3, 4);
-  EXPECT_EQ(inst.kind(), TypedInst::Kind);
+auto MakeInstWithNumberedFields(InstKind kind) -> Inst {
+  Inst inst = InstTestHelper::MakeInst(kind, Parse::Node(1), TypeId(2), 3, 4);
+  EXPECT_EQ(inst.kind(), kind);
   EXPECT_EQ(inst.parse_node(), Parse::Node(1));
   EXPECT_EQ(inst.type_id(), TypeId(2));
+  return inst;
+}
 
+template <typename TypedInst>
+auto CommonFieldOrder() -> void {
+  Inst inst = MakeInstWithNumberedFields(TypedInst::Kind);
```

Comment:
Shouldn't ExpectEqInsts be doing essentially the same comparisons that are happening below? That is, could this also be merged?

Reply:
This one is doing a comparison of the typed instruction's fields, whereas `ExpectEqInsts` is comparing the fields of an untyped instruction.